xdsl/transforms/csl_stencil_to_csl_wrapper.py

```diff
@@ -144,43 +144,64 @@ def _translate_function_args(
         """
         arg_ops: list[Operation] = []
         arg_op_mapping: list[SSAValue] = []
+        ptr_converts: list[Operation] = []
+        export_ops: list[Operation] = []
+        cast_ops: list[Operation] = []
+
         for arg in args:
             arg_name = arg.name_hint or ("arg" + str(args.index(arg)))
 
             if isa(arg.type, stencil.FieldType[TensorType[Attribute]]):
                 arg_t = csl_stencil_bufferize.tensor_to_memref_type(
                     arg.type.get_element_type()
                 )
-                arg_ops.append(
-                    memref.Global.get(
-                        sym_name=StringAttr(arg_name),
-                        sym_type=csl_stencil_bufferize.tensor_to_memref_type(
-                            arg.type.get_element_type()
-                        ),
-                        initial_value=UnitAttr(),
-                        sym_visibility=StringAttr("public"),
+                arg_ops.append(alloc := memref.Alloc([], [], arg_t))
+                ptr_converts.append(
+                    address := csl.AddressOfOp(
+                        operands=[alloc],
+                        result_types=[
+                            csl.PtrType(
+                                [
+                                    arg_t.get_element_type(),
+                                    csl.PtrKindAttr(csl.PtrKind.MANY),
+                                    csl.PtrConstAttr(csl.PtrConst.VAR),
+                                ]
+                            )
+                        ],
                     )
                 )
-                arg_ops.append(get_global_op := memref.GetGlobal(arg_name, arg_t))
-                arg_ops.append(
+                export_ops.append(csl.SymbolExportOp(arg_name, SSAValue.get(address)))
```

Comment:
Using the name hint can easily break. `SymbolExportOp` will export the symbol using the name given to it, but this is not necessarily the name the symbol will be printed with by the printer

Reply:
Is there any alternative?

Reply:
Some suggestions:
1. We could give `SymbolExportOp` the SSA value instead, so that in the printer it definitely uses the same name that the value will be printed with.
2. Another option is to give a symbol name to `csl.zeros` and `csl.constants`, but I don't really like that either.
3. A `csl.exported_array` (or something along those lines) which contains all functionality for defining and exporting arrays (we'd also then need some attribute on functions to make those exported).

All these options make it basically impossible to know the name of the symbol for interfacing from the host, but I think that's something to be resolved at some other time.

Option 3 is I think too much of a departure from the semantics of the CSL language (also a lot of work)

Option 2 would make creating new arrays needlessly difficult since not all arrays need to be exported (though I suppose we can make the symbol optional). This also complicates printing and does not escape from the problem of having to figure out what the symbol should be in the first place

Option 1 is the easiest to implement, but likely makes least sense from the point of view of the user accessing the exported symbols from the host.

I personally vote Option 1. I think it will cause the least number of changes and will be the easiest to implement. More suggestions are welcome :slightly_smiling_face: 

(I think this may have been one of the reasons I used a `memref.global` initially, but its hard to remember now :sweat_smile:)

Reply:
Option 1 sounds good - we actually do give it the symbol, iff it's a pointer rather than a function.  The one advantage of 'sym_name' properties is that they can be persisted when invoking mlir-opt, whereas name hints are all re-named.